ui/src/views/Clients/ViewClients/index.jsx

```diff
@@ -37,50 +42,36 @@ import ErrorPanel from '../../../components/ErrorPanel';
 export default class ViewClients extends PureComponent {
   state = {
     clientSearch: '',
-    // eslint-disable-next-line react/no-unused-state
-    previousClientId: '',
-    // This needs to be initially null in order for the defaultValue to work
-    value: null,
+    // This needs to be initially undefined in order for defaultValue to work
+    value: undefined,
   };
 
-  static getDerivedStateFromProps(props, state) {
-    // Any time the current user changes,
-    // Reset state to reflect new user / log out and default clientSearch query
-    if (
-      props.user &&
-      props.user.credentials.clientId !== state.previousClientId
-    ) {
-      return {
-        clientSearch: props.user.credentials.clientId,
-        previousClientId: props.user.credentials.clientId,
-      };
-    }
-
-    if (!props.user && state.previousClientId !== '') {
-      return {
-        clientSearch: '',
-        previousClientId: '',
-      };
-    }
-
-    return null;
-  }
-
-  handleClientSearchSubmit = clientSearch => {
+  handleClientSearchSubmit = async clientSearch => {
     const {
       data: { refetch },
     } = this.props;
+    const searchUri = this.props.history.location.search
+      ? decodeURIComponent(parse(this.props.history.location.search)['?search'])
+      : '';
 
     this.setState({ clientSearch });
 
-    refetch({
+    await refetch({
```

Comment:
Added in await to prevent scenario where user freezes client by refetching data that's already been fetched:

e.g.
1. Search for 'email'
2. Search another query with incorrect string e.g. 'sadsa asdas' -> without await, code will treat this as valid query and change the url query parameter
3. Search for 'email' again -> now possible even though we initially prevent searching for the same data twice
4. Client freezes

Reply:
> Search another query with incorrect string e.g. 'sadsa asdas' -> without await, code will treat this as valid query and change the url query parameter

That's okay. I think it should be safe to remove the `await` statement.

> Client freezes

Interesting. It seems to happen whether we `await` or not.  I can try to investigate. Any ideas?